libvast/builtins/formats/zeek_tsv.cpp

```diff
@@ -546,17 +445,152 @@ auto parser_impl(generator<std::optional<std::string_view>> lines,
     co_return;
   }
   auto metadata = zeek_metadata{};
-  auto parsed = metadata.parse_header(it, lines, ctrl);
-  if (not parsed) {
-    ctrl.abort(parsed.error());
-    VAST_ERROR("aborting Zeek metadata parsing");
-    co_return;
-  }
-  ++it;
-  auto closed = false;
-  auto b = table_slice_builder{metadata.temp_slice_schema};
-  std::vector<data> xs(metadata.fields.size());
+  auto sep_parser
+    = "#separator" >> ignore(+(parsers::space)) >> +(parsers::any);
+  auto sep_option = std::string{};
+  auto header_parsed = false;
+  auto closed = true;
+  auto b = std::optional<table_slice_builder>{};
+  auto xs = std::vector<data>{};
   for (; it != lines.end(); ++it) {
+    if (*it and (*it)->starts_with("#separator")) {
```

Comment:
I think from a readability perspective it would be much better to just move the nullopt check to the very top of the loop, i.e.,

```c++
for (; it != lines.end(); ++it) {
  auto current_line = *it;
  if (not current_line) {
    co_yield {};
    continue;
  }
  if (current_line->empty()) {
    continue;
  }
  // work with *current_line instead of **it
}
```

Reply:
Agreed. I added `current_line` in https://github.com/tenzir/tenzir/pull/3291/commits/d3dfa2a19440753d1d78858caf7286d7e849e792